rapids_build_backend/config.py

```diff
@@ -1,17 +1,30 @@
 # Copyright (c) 2024, NVIDIA CORPORATION.
 
 import os
+from typing import TYPE_CHECKING
 
 from .utils import _get_pyproject
 
+if TYPE_CHECKING:
+    from typing import Callable
+
+    # config options can be one of these types...
+    config_val_type = str | bool | None
+
+    # ... or a callable that returns one of those or a list of strings
+    mutable_config_val_type = Callable[[], config_val_type | list[str]]
```

Comment:
We might return mutable types other than lists of strings in the future. That's why I made `mutable_config_types` contain only the actual mutable types, and not the `Callable` itself, though I suppose it's just six of one and half a dozen of the other.

Reply:
Oh ok, I didn't understand that.

How about this?

https://github.com/rapidsai/rapids-build-backend/pull/19/commits/1873fc5d9c6950c3162855c3f3588289eb87b220

Reply:
Yes, that looks good.